xdsl/dialects/csl/csl.py

```diff
@@ -1445,6 +1446,29 @@ class SymbolExportOp(IRDLOperation):
 
     type = prop_def(base(PtrType) | base(FunctionType))
 
+    def __init__(self, sym_name: str | StringAttr, type_or_op: SSAValue | FunctionType):
+        """
+        Accepts one of:
+          1. var_name + PtrType + operand
+          2. var_name + FunctionType
+        """
+        var_name: StringAttr | SymbolRefAttr = (
```

Comment:
I don't really understand how this can be a `SymbolRefAttr`

Reply:
A few lines below there is a conversion to SymbolRefAttr. There is some complicated verification logic happening, that I sought to simplify in the constructor.